cli/azd/cmd/container.go

```diff
@@ -214,6 +215,35 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 			return env, nil
 		},
 	)
+	container.RegisterSingleton(func(
+		ctx context.Context,
+		console input.Console,
+		accountManager account.Manager,
+		userProfileService *azcli.UserProfileService,
+		subResolver account.SubscriptionTenantResolver,
+	) environment.EnvironmentResolver {
+		return func() (*environment.Environment, error) {
+			azdCtx, err := azdcontext.NewAzdContext()
+			if err != nil {
+				return nil, err
+			}
+			azdEnvs, err := azdCtx.ListEnvironments()
+			if err != nil {
+				return nil, err
+			}
+			var selectedEnv string
+			for _, env := range azdEnvs {
+				if env.IsDefault {
+					selectedEnv = env.Name
+				}
+			}
+			if selectedEnv == "" {
+				log.Panic("Found azd environments but none is selected.")
+			}
+			return loadOrInitEnvironment(
```

Comment:
I think this will trigger an interactive prompt flow, which might not be what the user would expect when running their code. Have you considered how we could mitigate this?